### PR TITLE
New style broadcast function

### PR DESCRIPTION
**Pull request by okuta**
Please see #3147.

### CODE REVIEW

File: tests/chainer_tests/functions_tests/array_tests/test_broadcast.py [hunk 1]
```diff
@@ -65,9 +64,10 @@ def test_forward_gpu(self):
         self.check_forward([cuda.to_gpu(x) for x in self.data])
 
     def check_backward(self, data, grads):
```
Comment (unnonouno) [inline tests/chainer_tests/functions_tests/array_tests/test_broadcast.py]: Can you add a test for double backward?

File: chainer/functions/array/broadcast.py [hunk 2]
```diff
@@ -2,31 +2,24 @@
...
-        # An input variable is always an array, not a scalar.
-        # We need to convert a scalar value to a zero-dim array.
-        if xp.isscalar(g):
-            g = xp.array(g)
-
-    axis = tuple(i for i, sx in enumerate(shape) if sx == 1)
-    if len(axis) > 0:
-        return g.sum(keepdims=True, axis=axis)
-    else:
+def _backward_one(g, shape):
```
Comment (unnonouno) [inline chainer/functions/array/broadcast.py]: Do we need to check if g is not None?

File: chainer/functions/array/broadcast.py [hunk 3]
```diff
@@ -2,31 +2,24 @@
...
+    if g.shape == shape:
         return g
+    ndim = len(shape)
+    lead = g.ndim - ndim
+    lead_axis = tuple(range(lead))
```
Comment (unnonouno) [inline chainer/functions/array/broadcast.py]: six.moves.range

File: chainer/functions/array/broadcast.py [hunk 4]
```diff
@@ -83,7 +74,7 @@ def broadcast(*args):
         True
 
     """
-    return Broadcast()(*args)
+    return Broadcast().apply(args)
```
[the remark this answers is not in the document]
Comment (okuta) [reply chainer/functions/array/broadcast.py]: I will fix

**Comment by unnonouno** [comment]
Check travis result

**Comment by okuta** [comment]
I fixed.

**Comment by unnonouno** [comment]
jenkins test this please

**Comment by unnonouno** [comment]
LGTM

**Pull request merged**